src/accelerate/utils/dataclasses.py

```diff
@@ -194,6 +228,11 @@ class FP8RecipeKwargs(KwargsHandler):
     override_linear_precision: Tuple[bool, bool, bool] = (False, False, False)
 
     def __post_init__(self):
+        self.backend = self.backend.upper()
+        if self.backend not in ["MSAMP", "TE"]:
+            raise ValueError("`backend` must be 'MSAMP' or 'TE' (TransformerEngine).")
+        if self.opt_level not in ["O1", "O2"]:
+            raise NotImplementedError("MS-AMP with Accelerate is only supported for `optimization_level` '01' or '02'")
```

Comment:
This error could be confusing when users use TE. Should the check include that the chosen backend is indeed MS-AMP?